File: cakk-api/src/main/java/com/cakk/api/dto/response/cake/CakeImageListResponse.java
package com.cakk.api.dto.response.cake;

import java.util.List;

import lombok.Builder;

import com.cakk.domain.dto.param.cake.CakeImageResponseParam;

@Builder
public record CakeImageListResponse(

	List<CakeImageResponseParam> cakeImages,
	Long lastCakeId,
	int size
) {

	public static CakeImageListResponse from(List<CakeImageResponseParam> cakeImages) {
		int size = cakeImages.size();

		return CakeImageListResponse.builder()
			.cakeImages(cakeImages)
			.lastCakeId(cakeImages.isEmpty() ? null : cakeImages.get(size - 1).cakeId())
			.size(cakeImages.size())
			.build();
	}
}

File: cakk-core/src/main/kotlin/com/cakk/core/dto/response/cake/CakeImageListResponse.kt
package com.cakk.core.dto.response.cake

import com.cakk.domain.mysql.dto.param.cake.CakeImageResponseParam

data class CakeImageListResponse(
	val cakeImages: List<CakeImageResponseParam>,
	val lastCakeId: Long?,
	val size: Int
)
